src/main/kotlin/org/rust/lang/core/psi/visitors/RustVisitorEx.kt

```diff
@@ -1,10 +1,19 @@
 package org.rust.lang.core.psi.visitors
 
+import com.intellij.psi.PsiFile
 import org.rust.lang.core.psi.RustLiteral
 import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustFile
 
 open class RustVisitorEx : RustVisitor() {
     open fun visitLiteral(literal: RustLiteral) = visitElement(literal)
     open fun visitNumericLiteral(literal: RustLiteral.Number) = visitLiteral(literal)
     open fun visitTextLiteral(literal: RustLiteral.Text) = visitLiteral(literal)
+    open fun visitRustFile(file: RustFile) {
+        super.visitFile(file)
+    }
+
+    override fun visitFile(file: PsiFile) {
+        if (file is RustFile) visitRustFile(file) else super.visitFile(file)
+    }
```

Comment:
>it could be easily removed and that if (file is RustFile) ... pattern could be used instead. 

I don't think that requiring `as` casts is a good API. It is much better to encapsulate them behind the functions (if it is not possible to eliminate them all together).

But we already have such encapsulating function (`rustMod`) and this visitor extension is indeed unnecessary, thanks for pointing this out! Will remove it.